enterprise/server/tasksize_model/tasksize_model.go

```diff
@@ -1,32 +1,309 @@
+	servingAddress     = flag.String("remote_execution.task_size_model.serving_address", "", "Address pointing to TF server with saved Tensorflow models (cpu, mem).")
 )
 
-type Model interface {
-	Predict(task *repb.ExecutionTask) *scpb.TaskSize
+const (
+	// minMemoryPredictionBytes is the minimum memory value to be returned
+	// from Predict().
+	minMemoryPredictionBytes = 1 * 1e6 // 1MB
+
+	// minMilliCPUPrediction is the minimum CPU value to be returned from
+	// Predict().
+	minMilliCPUPrediction = 1 // 1ms
+
+	// How much time to allow for initializing the model (connecting to the
+	// TF server and making a test prediction).
+	initTimeout = 10 * time.Second
+
+	// How long to wait before giving up on an individual prediction.
+	// This is intentionally very short because in the worst case where all
+	// requests are failing, this can become a bottleneck for RBE performance.
+	predictionTimeout = 5 * time.Millisecond
+
+	// Names referencing the saved model.
+
+	memModelName    = "mem"
+	cpuModelName    = "cpu"
+	modelInputName  = "x_input"
+	modelOutputName = "y"
+)
+
+var (
+	// fileExtensionRegexp matches file extensions. This is the same regexp
+	// used when training the model.
+	fileExtensionRegexp = regexp.MustCompile(`\.[a-z0-9]{1,4}$`)
+)
+
+type FeaturesConfig struct {
+	OSValues   []string `json:"os_values"`
+	ArchValues []string `json:"arch_values"`
+	Tools      []string `json:"tools"`
+	TestSizes  []string `json:"test_sizes"`
+	Extensions []string `json:"extensions"`
+}
+
+type Model struct {
+	config         *FeaturesConfig
+	extensionIndex map[string]int
+
+	client tf.PredictionServiceClient
 }
 
-func isPredictionEnabled(task *repb.ExecutionTask) bool {
+// New returns a TensorFlow model configured from the root path.
+func New(env environment.Env) (*Model, error) {
+	m := &Model{}
+
+	f, err := os.Open(*featuresConfigPath)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+	b, err := io.ReadAll(f)
+	if err != nil {
+		return nil, err
+	}
+	m.config = &FeaturesConfig{}
+	if err := json.Unmarshal(b, m.config); err != nil {
+		return nil, err
+	}
+	m.extensionIndex = make(map[string]int, len(m.config.Extensions))
+	for i, v := range m.config.Extensions {
+		m.extensionIndex[v] = i
+	}
+
+	// Connect to TF serving.
+	conn, err := grpc_client.DialTarget(*servingAddress)
+	if err != nil {
+		return nil, status.UnavailableErrorf("failed to dial TensorFlow serving at %s: %s", *servingAddress, err)
+	}
+	env.GetHealthChecker().RegisterShutdownFunction(func(ctx context.Context) error {
+		return conn.Close()
+	})
+	m.client = tf.NewPredictionServiceClient(conn)
+
+	// Run a test prediction to ensure the feature vector shape matches the
+	// model's expected input shape.
+	ctx := env.GetServerContext()
+	ctx, cancel := context.WithTimeout(ctx, initTimeout)
+	defer cancel()
+
+	x := m.featureVector(&repb.ExecutionTask{})
```

Comment:
Is this check going to block app startup? would be better to allow the app to startup w/ the model disabled

Reply:
I can update the PR that calls New() to log the returned error as log.Error instead of log.Fatal.